### PR TITLE
v0.13.x: Backport NumPy 1.14 compatibility

### PR DESCRIPTION
**Pull request by hmaarrfk**
* Fix numpy compatibility
* Use .14 build scripts to get .13 to compile.

Sorry for the messy git. I tried to force push as much as I could. Maybe I should just create a new branch and cleanup the git history.

I don't know why code coverage is failing.
## Description
[Tell us about your new feature, improvement, or fix! If relevant, please supplement the PR with images.]

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](http://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: doc/examples/filters/plot_cycle_spinning.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+"""
```
Comment (jni) [inline doc/examples/filters/plot_cycle_spinning.py]: @hmaarrfk did this file, cycle spinning, and hysteresis thresholding get caught in here by mistake? I don't think cycle spinning at least is in the 0.13 branch, though maybe I'm wrong...

Comment (hmaarrfk) [reply doc/examples/filters/plot_cycle_spinning.py]: Your right. I'm glad I'm not trying to backport the other big patch I submitted ;).

**Comment by hmaarrfk** [comment]
@jni no guarantee that it will work yet. I was kinda doing this while taking breaks. Learning how to cherry-pick from git.

**Comment by jni** [comment]
@hmaarrfk a valuable skill! =)

**Pull request merged**